core/src/main/java/aima/core/search/basic/support/BasicNode.java

```diff
@@ -42,14 +42,17 @@ public A action() {
 	public double pathCost() {
 		return pathCost;
 	}
-	
+
+	/**
+	 * @return json like representation of the node with its parents, recursively till root node
+	 */
 	@Override
 	public String toString() {
-		StringJoiner sj = new StringJoiner(", ", "Node(", ")");
-		sj.add(state.toString());
-		sj.add(parent == null ? "null" : parent.state().toString());
-		sj.add(action == null ? "null" : action.toString());
-		sj.add(""+pathCost);
+		StringJoiner sj = new StringJoiner(", ", "{", "}");
+		sj.add("parent: " + (parent == null ? "null" : parent.toString()));
+		sj.add("state: '" + state.toString() + "'");
+		sj.add("fromAction: " + (action == null ? "null" : "'" + action.toString() + "'"));
+		sj.add("withPathCost: "+pathCost);
 		return sj.toString();
 	}
 
```

Comment:
I do not wish to add JSON formatting of toStrings() within the core package. Ideally, an output specific format should be handled in a separate way.